src/xmldsig-validator.ts

```diff
@@ -0,0 +1,353 @@
+import { DOMParser } from "@xmldom/xmldom";
+import { SignedXml } from "./signed-xml";
+import {
+  KeySelectorFunction,
+  IdAttributeType,
+  HashAlgorithmMap,
+  TransformAlgorithmMap,
+  SignatureAlgorithmMap,
+  SignedXmlOptions,
+} from "./types";
+
+const DEFAULT_MAX_TRANSFORMS = 4;
+const DEFAULT_THROW_ON_ERROR = false;
+const DEFAULT_CHECK_CERT_EXPIRATION = true;
+
+export type CertificateKeySelector = {
+  /** Public certificate or key to use for validation */
+  publicCert: KeyLike;
+};
+
+export type KeyInfoKeySelector = {
+  /** Function to extract the public key from KeyInfo element */
+  getCertFromKeyInfo: (keyInfo?: Node | null) => string | null;
+};
+
+export type KeySelector = CertificateKeySelector | KeyInfoKeySelector;
+
+export interface XmlDsigValidatorSecurityOptions {
+  /**
+   * Maximum number of transforms allowed per Reference element.
+   * Limits complexity to prevent denial-of-service attacks.
+   * @default {@link DEFAULT_MAX_TRANSFORMS}
+   */
+  maxTransforms: number;
+
+  /**
+   * Check certificate expiration dates during validation.
+   * If true, signatures with expired certificates will be considered invalid.
+   * This only applies when using KeyInfoKeySelector
+   * @default true
+   */
+  checkCertExpiration: boolean;
+
+  /**
+   * Optional truststore of trusted certificates
+   * When provided, the certificate used to sign the XML must chain to one of these trusted certificates.
+   * These must be PEM or DER encoded X509 certificates
+   */
+  truststore?: Array<string | Buffer | X509Certificate>;
+
+  /**
+   * Signature algorithms allowed during validation.
+   *
+   * @default {@link SignedXml.getDefaultSignatureAlgorithms()}
+   */
+  signatureAlgorithms?: SignatureAlgorithmMap;
+
+  /**
+   * Hash algorithms allowed during validation.
+   *
+   * @default {@link SignedXml.getDefaultHashAlgorithms()}
+   */
+  hashAlgorithms?: HashAlgorithmMap;
+
+  /**
+   * Transform algorithms allowed during validation. (This must include canonicalization algorithms)
+   *
+   * @default all algorithms in {@link SignedXml.getDefaultTransformAlgorithms()}
+   */
+  transformAlgorithms?: TransformAlgorithmMap;
+
+  /**
+   * Canonicalization algorithms allowed during validation.
+   *
+   * @default all algorithms in {@link SignedXml.getDefaultCanonicalizationAlgorithms()}
+   */
+  canonicalizationAlgorithms?: TransformAlgorithmMap;
+}
+
+/**
+ * Common configuration options for XML-DSig validation.
+ */
+interface XmlDSigValidatorOptions {
+  /**
+   * Key selector for determining the public key to use for validation.
+   */
+  keySelector: KeySelector;
+
+  /**
+   * Names of XML attributes to treat as element identifiers.
+   * Used when resolving URI references in signatures.
+   * When passing strings, only the localName is matched, ignoring namespace.
+   * To explicitly match attributes without namespaces, use: { localName: "Id", namespaceUri: undefined }
+   * @default ["Id", "ID", "id"]
```

Comment:
I worry about specifying defaults this way because code drift often causes these detached comments to become out-dated. I'd just prefer to keep this in a README, which I suppose can be linked to from here.